metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -132,38 +132,50 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 	}
 
 	for _, vm := range vmt {
-		freeMemory := (int64(vm.Summary.Config.MemorySizeMB) * 1024 * 1024) - (int64(vm.Summary.QuickStats.GuestMemoryUsage) * 1024 * 1024)
-
+		usedMemory := int64(vm.Summary.QuickStats.GuestMemoryUsage) * 1024 * 1024
+		usedCPU := vm.Summary.QuickStats.OverallCpuUsage
 		event := common.MapStr{
 			"name": vm.Summary.Config.Name,
 			"os":   vm.Summary.Config.GuestFullName,
 			"cpu": common.MapStr{
 				"used": common.MapStr{
-					"mhz": vm.Summary.QuickStats.OverallCpuUsage,
+					"mhz": usedCPU,
 				},
 			},
 			"memory": common.MapStr{
 				"used": common.MapStr{
 					"guest": common.MapStr{
-						"bytes": (int64(vm.Summary.QuickStats.GuestMemoryUsage) * 1024 * 1024),
+						"bytes": usedMemory,
 					},
 					"host": common.MapStr{
 						"bytes": int64(vm.Summary.QuickStats.HostMemoryUsage) * 1024 * 1024,
 					},
 				},
-				"total": common.MapStr{
-					"guest": common.MapStr{
-						"bytes": int64(vm.Summary.Config.MemorySizeMB) * 1024 * 1024,
-					},
-				},
-				"free": common.MapStr{
-					"guest": common.MapStr{
-						"bytes": freeMemory,
-					},
-				},
 			},
 		}
 
+		totalCPU := vm.Summary.Config.CpuReservation
+		if totalCPU > 0 {
```

Comment:
Issue created to be able to test with the simulator for different cases of reported metrics https://github.com/elastic/beats/issues/26287